api/peeropts.go

```diff
@@ -0,0 +1,22 @@
+// Copyright © 2023 Attestant Limited.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+// PeerOpts are the options for client side peer filtering.
+type PeerOpts struct {
+	// State of the connection (disconnected, connecting, connected, disconnecting)
+	State []string
```

Comment:
This and `Direction` would be better off as enums (as much as we can make them).

Reply:
To do this would break the peeropts(api) / peer(v1) boundary. We can slide the peeropts into the v1 but that kind of sidesteps my interpretation of the refactor.

Reply:
We can leave it for now.